### PR TITLE
(PC-25783) fix(emailApp): sentry error unable to open url message

### PR DESCRIPTION
**Pull request by vikingprank**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-25783
### Error 100% iOS
Hypothesis: Info.plist was missing the necessary configuration for our app to be able to open any other mail app than Apple Mail. 

### Why were we getting the error?
It seems like we are trying to open the Apple mail app, but it is unavailable/not found...

### But why are we trying to open the app when it doesn't exist?

When I "force" opening apple mail with `await openInbox({ app: 'apple-mail' })`, and press the button `Consulter mes mails` on a physical phone, I am able to reproduce the error message `Unable to open URL: message://`.
When I try to use only `await openInbox()`, the phone tries to open apple-mail even though I have the gmail app on the phone and it is set as default mail.

### Maybe the user had deleted the default mail app?
This is the only explanation I can come up with, but it might be wrong since it seems surprising that so many users (14k) would deleted the apple mail app.

### An additional explanation:
The error was first seen Sep 14th 2023, 5:50 AM, 3 months ago.
It is weird that this issue started coming up much more recently, even though we have been using the lib `react-native-email-link` for the last 2 years. 
Maybe there was some kind of recent update on iOS that changed the behavior of the mail app?

It might be unlikely, but it is possible. I'm hoping the error will end up disappearing with the addition of code in info.plist

## Checklist

I have:

- [x] Made sure my feature is working on Victor's iPhone
- [x] The device should now show a list of available mail apps when clicked "Consulter mes mails" (do the forgot password process)
- [x] If there is an issue opening mail apps, there should be a snack bar explaining to the user to open their mail themselves

image

### CODE REVIEW

File: ios/PassCulture/Info.plist [hunk 1]
```diff
@@ -73,6 +73,19 @@
...
 		<string>viber</string>
 		<string>whatsapp</string>
+		<string>mailto</string>
+		<string>message</string>
+		<string>readdle-spark</string>
+		<string>airmail</string>
+		<string>ms-outlook</string>
+		<string>googlegmail</string>
+		<string>inbox-gmail</string>
+		<string>ymail</string>
```
Comment (voisinhugo) [inline ios/PassCulture/Info.plist:83]: On dirait qqun qui a mal écrit gmail 😆

**Review by voisinhugo — APPROVED** [review]

File: src/features/auth/components/OpenInboxButton.tsx [hunk 2]
```diff
@@ -2,13 +2,24 @@ import React from 'react'
...
   const onPress = async () => {
-    await openInbox()
-    onAdditionalPress?.()
+    try {
+      await openInbox()
+      onAdditionalPress?.()
+    } catch (error) {
+      showErrorSnackBar({
+        message:
+          'Nous n’avons pas réussi à ouvrir votre boîte e-mail. Veuillez l’ouvrir manuellement.',
```
Comment (voisinhugo) [inline src/features/auth/components/OpenInboxButton.tsx:18]: Le wording a bien été vu avec PM ou UX ?

Comment (vikingprank) [reply src/features/auth/components/OpenInboxButton.tsx:18]: On en a parlé vite fait avec Victor, ça doit encore être discuté avec Mouna!

**Comment by vikingprank** [comment]
We are now hiding the button on iOS when no mail boxes are available:
Screenshot 2023-12-01 at 10 37 21
Left Android / Right ios simulator (it isn't possible to install mail apps on simulator)

File: src/features/auth/components/EmailSentGeneric.tsx [hunk 3]
```diff
@@ -28,6 +30,14 @@ export const EmailSentGeneric: FunctionComponent<Props> = ({
...
   openInBoxAnalytics,
 }) => {
+  const isMailAppAvailable = useIsMailAppAvailableIOS()
+  const shouldDisplayOpenInboxButton = () => {
+    if (Platform.OS === 'ios') {
+      return isMailAppAvailable
+    } else {
+      return true
+    }
+  }
```
Comment (aliraiki) [inline src/features/auth/components/EmailSentGeneric.tsx:40]:
```suggestion
  const shouldDisplayOpenInboxButton = () => {
      return Platform.OS === 'ios' ? isMailAppAvailable : true
  }
```

File: src/features/bookings/components/TicketBody/EmailSent/EmailSent.tsx [hunk 4]
```diff
@@ -14,10 +15,16 @@ type Props = {
 
 export const EmailSent: FunctionComponent<Props> = ({ offerDate }) => {
   const emailMessage = getEmailMessage(offerDate)
+  const isMailAppAvailable = useIsMailAppAvailableIOS()
+  const shouldDisplayOpenInboxButton = () => {
+    if (Platform.OS === 'ios') return isMailAppAvailable
+    if (Platform.OS === 'web') return false
+    else return true
```
Comment (aliraiki) [inline src/features/bookings/components/TicketBody/EmailSent/EmailSent.tsx:22]:
```suggestion
    return true
```

File: src/features/profile/components/Badges/SubscriptionMessageBadge.tsx [hunk 5]
```diff
@@ -33,10 +36,15 @@ const CallToAction = ({ subscriptionMessage }: Props) => {
     inline: true as BaseButtonProps['inline'],
   }
 
+  const shouldDisplayOpenInboxButton = () => {
+    if (Platform.OS === 'ios') return isMailAppAvailable
+    else return true
```
Comment (aliraiki) [inline src/features/profile/components/Badges/SubscriptionMessageBadge.tsx:41]:
```suggestion
    return Platform.OS === 'ios' ? isMailAppAvailable : true
```

**Review by aliraiki — APPROVED** [review]

File: src/features/bookings/components/TicketBody/EmailSent/EmailSent.native.test.tsx [hunk 6]
```diff
@@ -5,10 +5,24 @@ import { render, screen } from 'tests/utils'
...
+
 describe('<EmailSent/>', () => {
   it('should display the button "Consulter mes e-mails"', () => {
     render(<EmailSent offerDate={offerDate} />)
 
     expect(screen.queryByTestId('Consulter mes e-mails')).toBeOnTheScreen()
   })
+
+  it('should not show the button to open mail if no mail app is available', async () => {
+    mockIsMailAppAvailable = false
```
Comment (aliraiki) [inline src/features/bookings/components/TicketBody/EmailSent/EmailSent.native.test.tsx:21]: Même remarque par rapport à l'indépendance des tests

File: src/features/bookings/components/TicketBody/TicketBody.native.test.tsx [hunk 7]
```diff
@@ -78,5 +83,24 @@ describe('TicketBody', () => {
...
+      it('should show the button to open mail', async () => {
+        render(<TicketBody {...initialProps} withdrawalType={undefined} />)
+
+        const checkEmailsButton = screen.queryByText('Consulter mes e-mails')
+
+        expect(checkEmailsButton).toBeNull()
+      })
+
+      it('should not show the button to open mail if no mail app is available', async () => {
+        mockIsMailAppAvailable = false
```
Comment (aliraiki) [inline src/features/bookings/components/TicketBody/TicketBody.native.test.tsx:97]: Même remarque par rapport à l'indépendance des tests

File: src/features/auth/pages/forgottenPassword/ResetPasswordEmailSent/ResetPasswordEmailSent.native.test.tsx [hunk 8]
```diff
@@ -18,6 +23,7 @@ const routeMock: RouteProp<RootStackParamList, 'ResetPasswordEmailSent'> = {
 
 describe('<ResetPasswordEmailSent />', () => {
   it('should match snapshot', () => {
+    mockIsMailAppAvailable = true
```
[the remark this answers is not in the document]
Comment (vikingprank) [reply src/features/auth/pages/forgottenPassword/ResetPasswordEmailSent/ResetPasswordEmailSent.native.test.tsx:26]: à merci je vais faire ça !

**Pull request merged**